app/src/main/java/com/realityexpander/tasky/presentation/splash_screen/SplashViewModel.kt

```diff
@@ -0,0 +1,84 @@
+package com.realityexpander.tasky.presentation.splash_screen
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.dataStore
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.realityexpander.tasky.TaskyApplication
+import com.realityexpander.tasky.common.settings.AppSettings
+import com.realityexpander.tasky.common.settings.AppSettingsSerializer
+import com.realityexpander.tasky.data.repository.remote.IAuthApi
+import com.realityexpander.tasky.domain.AuthInfo
+import com.realityexpander.tasky.domain.IAuthRepository
+import com.realityexpander.tasky.presentation.common.UIConstants
+import com.realityexpander.tasky.presentation.common.util.UiText
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.yield
+import javax.inject.Inject
+
+@HiltViewModel
+class SplashViewModel @Inject constructor(
+    private val authRepository: IAuthRepository,
+    private val savedStateHandle: SavedStateHandle,
+) : ViewModel() {
+
+    private val authInfo: AuthInfo? =
+        savedStateHandle[UIConstants.SAVED_STATE_authInfo]
+    private val statusMessage: UiText =
+        savedStateHandle[UIConstants.SAVED_STATE_statusMessage] ?: UiText.None
+
+    private val _splashState = MutableStateFlow(SplashState())
+    val splashState = _splashState.onEach { state ->
+        // save state for process death // is this needed for splash screen?
+        savedStateHandle[UIConstants.SAVED_STATE_authInfo] = state.authInfo
+        savedStateHandle[UIConstants.SAVED_STATE_statusMessage] = state.statusMessage
+
+        println("SplashViewModel: splashState: $state")
+    }.stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), SplashState())
+
+    init {
+        viewModelScope.launch {
+            yield() // allow the splashState to be initialized
+
+            // restore state after process death -- // is this needed for splash screen?
+            _splashState.update {
+                it.copy(
+                    authInfo = authInfo,
+                    authInfoChecked = false,
+                    statusMessage = statusMessage,
+                )}
+            yield() // allow the splashState to be restored
+        }
+    }
+
+    fun onSetAuthInfo(authInfo: AuthInfo) {
+        viewModelScope.launch {
+
+            // set the AuthInfo and AuthToken for this user
+            TaskyApplication.authInfoGlobal = authInfo
+            IAuthApi.setAuthToken(authInfo.authToken)
```

Comment:
Here, you directly access a data layer dependency from presentation which creates coupling between these two layers and makes them dependent of each other. That's one thing we aim to avoid with a layered architecture, so this should rather be done in the use case or repo directly after logging in.

Reply:
I will make an interface call thru the repository for this.

Thanks for pointing this out, im still getting used the multi-layered architecture pattern. I am so used to doing things directly, and old habits die hard lol.